From 750d2a9390a4e1af97e987fb915c13d0ec99acf0 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Wed, 19 Oct 2022 17:09:31 -0700
Subject: [PATCH] Revert "mm: align larger anonymous mappings on THP
 boundaries"

This reverts commit f35b5d7d676e59e401690b678cd3cfec5e785c23.

This causes a massive run time regression on my Thelio workstation in
the form of longer Linux kernel build times.

Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 mm/mmap.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/mm/mmap.c b/mm/mmap.c
index bf2122af94e7..158ce1c9abdf 100644
--- a/mm/mmap.c
+++ b/mm/mmap.c
@@ -1778,9 +1778,6 @@ get_unmapped_area(struct file *file, unsigned long addr, unsigned long len,
 		 */
 		pgoff = 0;
 		get_area = shmem_get_unmapped_area;
-	} else if (IS_ENABLED(CONFIG_TRANSPARENT_HUGEPAGE)) {
-		/* Ensures that larger anonymous mappings are THP aligned. */
-		get_area = thp_get_unmapped_area;
 	}
 
 	addr = get_area(file, addr, len, pgoff, flags);
-- 
2.38.0

